### ISSUE TITLE
Refactor init()

### ISSUE THREAD

**Issue by i-stam** [lukso-network/rICO-smart-contracts#9]
```solidity
uint256 lastStageBlockEnd = stage0.endBlock; => uint256 newStageBlockStart = stage0.endBlock + 1;  ??


buyPhaseBlockCount = lastStageBlockEnd - buyPhaseStartBlock; => buyPhaseBlockCount = lastStageBlockEnd - buyPhaseStartBlock + 1; 

e.g. `start block: 1, end block: 12,  blockCount = 12 - 1 +1 = 12;

```

**Comment by mickys** [comment]
Did not understand what you mean by this.
There's a stage 0 ( where participants buy in at initial price, and have no unlocked tokens), and 12 more stages.

**Comment by frozeman** [comment]
I made your comment more readable, but i dont know either what you propose.

Please lets make sure to use \`\`\`solidity ... \`\`\` for wrap code for readability.

**Comment by i-stam** [comment]
The _commitPhaseStartBlock should be checked against the current block in case it is set in the past!

**Comment by i-stam** [comment]
`uint256 lastStageBlockEnd = stage0.endBlock; `
https://github.com/lukso-network/rICO-smart-contracts/blob/develop/contracts/ReversibleICO.sol#L257
can be changed to something like `newStageBlockStart = stage0.endBlock + 1; ` for reducing operations.

**Comment by frozeman** [comment]
As this is only in the initialiser function its probably ok if its not super efficient.

**Comment by i-stam** [comment]
Addressed in #29
